src/libertem/udf/base.py

```diff
@@ -33,6 +33,7 @@
 from libertem.executor.base import Environment, JobExecutor
 
 if TYPE_CHECKING:
+    from typing import OrderedDict
```

Comment:
Btw. for using many types that are added in a newer Python version, we also depend on the "polyfill" `typing_extensions`, which is especially needed if you want to actually instantiate such a type using the same import (vs. having to import the corresponding type from `collections` in addition, for example)

Reply:
I tried, but didn't get that to work any better than it is now. I hope that's ok!

Reply:
That threw an error when I tested with Python 3.6, unfortunately. But I am not sure if I did sth wrong, maybe.